File: conexiones/conexion.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class conexion {
    public static void main(String[] args) {
        String url = "jdbc:mysql://localhost:3306/"; // Ajusta la URL si es necesario
        String user = "root";
        String password = "";
        
        try {
            Connection conn = DriverManager.getConnection(url, user, password);
            System.out.println("Conexion exitosa");
            conn.close();
        } catch (SQLException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

File: conexiones/conexion.go
package main

import (
	"database/sql"
	"fmt"
	_ "github.com/go-sql-driver/mysql"
)

func main() {
	// DSN: usuario:contraseña@tcp(host:puerto)/nombre_basedatos
	dsn := "root:@tcp(127.0.0.1:3306)/tienda" // Ajusta según tus datos
	db, err := sql.Open("mysql", dsn)
	if err != nil {
		fmt.Println("Error al abrir la conexión:", err)
		return
	}
	defer db.Close()

	err = db.Ping()
	if err != nil {
		fmt.Println("Error de conexión:", err)
		return
	}
	fmt.Println("Conexión exitosa a MySQL")
}
